src/karmabot/db/database.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import sys
+from contextlib import contextmanager
+
+import sqlalchemy as sa
+import sqlalchemy.orm as orm
+from sqlalchemy.exc import OperationalError
+
+from karmabot import settings
+from karmabot.db.modelbase import SqlAlchemyBase
+from karmabot.exceptions import NotInitializedException
+
+
+class Database:
+    """Class for handlading the database connection and provide access to sessions"""
+
+    def __init__(
+        self, connection_string: str = settings.DATABASE_URL, echo: bool = False
+    ) -> None:
+
+        self.connection_string = connection_string
+        self.echo = echo
+
+        self._engine = None
+        self._SessionFactory = None  # noqa
```

Comment:
`self._session`?

Reply:
All solved via the SQL Alchemy SessionFactory and `@props`